src/components/BigDataDropdown/index.tsx

```diff
@@ -0,0 +1,196 @@
+'use client';
+
+import { useCallback, useId, useMemo } from 'react';
+import { AsyncPaginate, LoadOptions } from 'react-select-async-paginate';
+import { GroupBase, MultiValue, SingleValue } from 'react-select';
+import { DropdownOption } from '@/types/dropdown';
+import { DropdownStyles, DropdownWrapper } from '../InputDropdown/styles';
+import { ErrorText, InputLabel } from '../TextInput/styles';
+import { AnimatedMenu, NoOptionsMessage } from '../InputDropdown';
+
+// for map: key is actual data stored, value is displayed
+interface CommonProps {
+  options: Set<string> | Map<string, string>;
+  label: string;
+  placeholder?: string;
+  error?: string;
+  disabled?: boolean;
+  required?: boolean;
+  pageSize?: number;
+}
+
+interface MultiSelectProps extends CommonProps {
+  multi: true;
+  onChange?: (value: Set<string>) => void;
+}
+
+interface SingleSelectProps extends CommonProps {
+  multi?: false;
+  onChange?: (value: string | null) => void;
+}
+
+type BigDataDropdownProps = SingleSelectProps | MultiSelectProps;
+
+export default function BigDataDropdown({
+  options,
+  label,
+  placeholder = '',
+  error = '',
+  disabled,
+  required,
+  multi,
+  pageSize = 10,
+  onChange,
+}: BigDataDropdownProps) {
+  const optionsArray = useMemo(
+    () =>
+      options instanceof Set
+        ? Array.from(options).map(v => ({ label: v, value: v }))
+        : Array.from(options.entries()).map(([k, v]) => ({
+            value: k,
+            label: v,
+          })),
+    [options],
+  );
+
+  const handleChange = useCallback(
```

Comment:
can we make it so that when something is typed into the search bar, we auto-scroll to the top? (not sure if this is the correct functionality? cc @davidqing6432)

Reply:
i think it makes sense, typically the most accurate results float to the top

Reply:
i implemented this feature for both the `BigDataDropdown` and the `InputDropdown`, but will leave this open for david's comments

Reply:
sounds good!!